lib/PoolCluster.js

```diff
@@ -124,17 +128,35 @@ PoolCluster.prototype._getNode = function(id) {
 };
 
 PoolCluster.prototype._increaseErrorCount = function(node) {
+  var self = this;
+
   if (++node.errorCount >= this._removeNodeErrorCount) {
     var index = this._serviceableNodeIds.indexOf(node.id);
     if (index !== -1) {
       this._serviceableNodeIds.splice(index, 1);
-      delete this._nodes[node.id];
-
       this._clearFindCaches();
 
-      node.pool.end();
+      if ((this._restoreNodeTimeout > 0) && !node.restoreTimeoutID) {
+        node.online = false;
+        this.emit('offline', node.id);
+
+        node.restoreTimeoutID = setTimeout(function() {
```

Comment:
Do we even need to do this using a timer? Why can we not calculate the restoration time and in functions that get nodes, simply query that? (and if `online` is important, it can be a getter). We may also want to use `process.hrtime` instead of dates and timers such that we are using a monotonically-increasing source that is not subject to system date/time adjustments.